lib/packwerk/run_context.rb

```diff
@@ -75,15 +75,29 @@ def initialize(
 
     sig { params(relative_file: String).returns(T::Array[Packwerk::Offense]) }
     def process_file(relative_file:)
+      reference_checker = ReferenceChecking::ReferenceChecker.new(@checkers)
+
+      references_result = references_from_file(relative_file: relative_file)
+
+      references_result.file_offenses +
+        references_result.references.flat_map { |reference| reference_checker.call(reference) }
+    end
+
+    class FileReferencesResult < T::Struct
```

Comment:
I believe T::Struct is slower than a normal struct. Please provide a benchmark, or switch to a plain old Struct (I think we only use Struct in other files).

Reply:
There are 4 other uses of `T::Struct` around the codebase and 4 uses of `Struct`.

I did not know about performance differences and chose `T::Struct` because it's typed, and I appreciate the added documentation and type safety, especially within this class that is a little messy and doesn't have great tests.

Reply:
A quick benchmark shows that access is about the same, while instantiation is ~5x slower for `T::Struct`. I was still able to instantiate 380k `T::Struct` within a second on my laptop.